### PR TITLE
Integration test for CollectTargetedPcrMetrics

### PR DESCRIPTION
**Pull request by dekling**

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetrics_IT.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.broadinstitute.hellbender.tools.IntegrationTestSpec;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Created by dkling on 9/1/15.
+ */
+public final class CollectTargetedPcrMetrics_IT extends CommandLineProgramTest {
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/directed/CollectTargetedPcrMetrics_IT.java]: rename this class to the proper name

**Comment by akiezun** [comment]
please clean up the pull request - include only changes relevant to this integration test.

Also, a test is failing on travis `Cannot read non-existent file: /home/travis/build/broadinstitute/hellbender/src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectBaseDistributionByCycle/first5000a.bam`

please fix those and reassign to me
back to @dekling

File: src/main/java/org/broadinstitute/hellbender/engine/ReferenceFileSource.java [hunk 2]
```diff
@@ -3,10 +3,8 @@
 import htsjdk.samtools.SAMSequenceDictionary;
 import htsjdk.samtools.reference.ReferenceSequence;
 import org.broadinstitute.hellbender.exceptions.GATKException;
-import org.broadinstitute.hellbender.utils.SimpleInterval;
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/engine/ReferenceFileSource.java]: remove this file from the branch

File: src/main/java/org/broadinstitute/hellbender/engine/ReferenceMemorySource.java [hunk 3]
```diff
@@ -2,15 +2,11 @@
 
 import htsjdk.samtools.SAMSequenceDictionary;
 import htsjdk.samtools.reference.ReferenceSequence;
-import org.broadinstitute.hellbender.exceptions.GATKException;
 import org.broadinstitute.hellbender.utils.SimpleInterval;
 import org.broadinstitute.hellbender.utils.Utils;
-import org.broadinstitute.hellbender.utils.fasta.CachingIndexedFastaSequenceFile;
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/engine/ReferenceMemorySource.java]: remove this file from the branch

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java [hunk 4]
```diff
@@ -20,7 +20,6 @@
 
 import java.io.File;
 import java.io.FileNotFoundException;
-import java.util.function.UnaryOperator;
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java]: remove this file from the branch

File: src/test/java/org/broadinstitute/hellbender/utils/read/ReadUtilsUnitTest.java [hunk 5]
```diff
@@ -1,9 +1,7 @@
 package org.broadinstitute.hellbender.utils.read;
 
-import htsjdk.samtools.Cigar;
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/utils/read/ReadUtilsUnitTest.java]: remove this file from the branch

**Comment by akiezun** [comment]
remove these file from the branch:
src/test/resources/org/broadinstitute/hellbender/metrics/first5000a.bai
src/test/resources/org/broadinstitute/hellbender/metrics/first5000a.bam

**Comment by akiezun** [comment]
remove this file MM.txt.alignment_summary_metrics

File: src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.alignment_summary_metrics [hunk 6]
```diff
@@ -0,0 +1,12 @@
+## htsjdk.samtools.metrics.StringHeader
+# picard.analysis.CollectMultipleMetrics INPUT=/Users/dkling/Programs/sandbox/firstthree/input/first5000a.bam OUTPUT=/Users/dkling/Programs/sandbox/firstthree/expected/MM.txt REFERENCE_SEQUENCE=/Users/dkling/Programs/sandbox/firstthree/input/ReferenceSequence/human_b37_20.fasta    ASSUME_SORTED=true STOP_AFTER=0 PROGRAM=[CollectAlignmentSummaryMetrics, CollectBaseDistributionByCycle, CollectInsertSizeMetrics, MeanQualityByCycle, QualityScoreDistribution] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
```
Comment (akiezun) [inline src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.alignment_summary_metrics]: remove this file form the branch

File: src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.base_distribution_by_cycle_metrics [hunk 7]
```diff
@@ -0,0 +1,211 @@
+## htsjdk.samtools.metrics.StringHeader
+# picard.analysis.CollectMultipleMetrics INPUT=/Users/dkling/Programs/sandbox/firstthree/input/first5000a.bam OUTPUT=/Users/dkling/Programs/sandbox/firstthree/expected/MM.txt REFERENCE_SEQUENCE=/Users/dkling/Programs/sandbox/firstthree/input/ReferenceSequence/human_b37_20.fasta    ASSUME_SORTED=true STOP_AFTER=0 PROGRAM=[CollectAlignmentSummaryMetrics, CollectBaseDistributionByCycle, CollectInsertSizeMetrics, MeanQualityByCycle, QualityScoreDistribution] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
```
Comment (akiezun) [inline src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.base_distribution_by_cycle_metrics]: remove this file form the branch

File: src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.insert_size_metrics [hunk 8]
```diff
@@ -0,0 +1,559 @@
+## htsjdk.samtools.metrics.StringHeader
```
Comment (akiezun) [inline src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.insert_size_metrics]: remove this file form the branch

File: src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.quality_by_cycle_metrics [hunk 9]
```diff
@@ -0,0 +1,211 @@
+## htsjdk.samtools.metrics.StringHeader
```
Comment (akiezun) [inline src/test/resources/org/broadinstitute/hellbender/tools/picard/analysis/CollectMultipleMetrics/MM.txt.quality_by_cycle_metrics]: remove this file form the branch

**Comment by akiezun** [comment]
remove the human_b37_20.fasta file from the branch - this file is too large. use another part of the reference

**Comment by akiezun** [comment]
back to @dekling - please remove all files that are not relevant to this test. Also, use a different reference (and maybe input files) because the one you used is too large.

File: src/main/java/org/broadinstitute/hellbender/engine/ReferenceFileSource.java (hunk 2 above)
Comment (dekling) [inline src/main/java/org/broadinstitute/hellbender/engine/ReferenceFileSource.java]: This does not exist.  There is a file called: import org.broadinstitute.hellbender.utils.Utils;.  Is this the one I should remove?

Comment (dekling) [inline src/main/java/org/broadinstitute/hellbender/engine/ReferenceFileSource.java]: import org.broadinstitute.hellbender.utils.iterators.ByteArrayIterator; does not seem to be present either.  I just see:

package org.broadinstitute.hellbender.engine;

import htsjdk.samtools.SAMSequenceDictionary;
import htsjdk.samtools.reference.ReferenceSequence;
import org.broadinstitute.hellbender.exceptions.GATKException;
import org.broadinstitute.hellbender.utils.Utils;
import org.broadinstitute.hellbender.utils.fasta.CachingIndexedFastaSequenceFile;

import java.io.File;
import java.io.IOException;
import java.util.Iterator;

Please let me know how I can resolve these discrepancies.

**Comment by dekling** [comment]
Is this OK now?

**Comment by akiezun** [comment]
@dekling are you on top of this or do you need help?

**Comment by dekling** [comment]
Happy to finish it when I get back.  However, feel free to take it over if you cannot wait.  I will be back on 10/13/15.
-D

**Comment by dekling** [comment]
Will leave this be.  See new pr #996.

**Pull request closed without merging**